packages/debugger/src/model.ts

```diff
@@ -22,12 +23,17 @@ export class DebuggerModel implements IDebugger.Model.IService {
   /**
    * Instantiate a new DebuggerModel
    */
-  constructor() {
-    this.breakpoints = new BreakpointsModel();
-    this.callstack = new CallstackModel();
+  constructor(
+    config: IDebugger.IConfig,
+    notebookTracker: INotebookTracker | null
+  ) {
+    this.breakpoints = new BreakpointsModel(config, notebookTracker);
+    this.callstack = new CallstackModel(config, notebookTracker);
     this.variables = new VariablesModel();
     this.sources = new SourcesModel({
-      currentFrameChanged: this.callstack.currentFrameChanged
+      currentFrameChanged: this.callstack.currentFrameChanged,
+      notebookTracker: notebookTracker,
+      config: config
```

Comment:
Let's use IOptions object, all models are part of public API and IOptions allows us to add/deprecate/remove arguments easily between versions

```suggestion
  constructor(options: DebuggerModel.IOptions) {
    this.breakpoints = new BreakpointsModel(config, notebookTracker);
    this.callstack = new CallstackModel(config, notebookTracker);
this.variables = new VariablesModel();
this.sources = new SourcesModel({
      currentFrameChanged: this.callstack.currentFrameChanged,
      notebookTracker: options.notebookTracker,
      config: options.config
```

Reply:
Thanks, Done in [1992584](https://github.com/jupyterlab/jupyterlab/pull/17881/commits/1992584d9655e62504ede06f342e792a9a55c548)